File: src/test/java/dev/retrotv/crypt/owe/sha/SHA384Test.java
package dev.retrotv.crypt.owe.sha;

import dev.retrotv.crypt.owe.OWETest;
import dev.retrotv.enums.HashAlgorithm;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class SHA384Test extends OWETest {

    @Test
    @DisplayName("SHA384 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(HashAlgorithm.SHA384);
    }

    @Test
    @DisplayName("SHA384 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new SHA384(), HashAlgorithm.SHA384);
    }

    @Test
    @DisplayName("SHA384 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new SHA384());
    }

    @Test
    @DisplayName("SHA384 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new SHA384());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA384Test.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class SHA384Test : OWETest() {
    @Test
    @DisplayName("SHA384 File hash 테스트")
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.SHA384)
    }

    @Test
    @DisplayName("SHA384 File hash matches 테스트")
    fun fileHashMatchesTest() {
        fileHashMatchesTest(SHA384(), Algorithm.Hash.SHA384)
    }

    @Test
    @DisplayName("SHA384 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(SHA384())
    }
}
